partialWaveFit/fitResult.cc

```diff
@@ -569,14 +569,17 @@ double
 fitResult::intensityErr(const string& waveNamePattern) const
 {
 	// get amplitudes that correspond to wave name pattern
-	const vector<unsigned int> prodAmpIndices = prodAmpIndicesMatchingPattern(waveNamePattern);
-	const unsigned int         nmbAmps        = prodAmpIndices.size();
-	if (not _covMatrixValid or (nmbAmps == 0)) {
+	const vector<unsigned int> waveIndices    = waveIndicesMatchingPattern(waveNamePattern);
+	const unsigned int         nmbWaves       = waveIndices.size();
+	if (not _covMatrixValid or (nmbWaves == 0)) {
 		return 0.;
 	}
-	if(nmbAmps == 1) {
-		return intensityErr(prodAmpIndices[0]);
+	if(nmbWaves == 1) {
+		return intensityErr(waveIndices[0]);
 	}
+
+	const vector<unsigned int> prodAmpIndices = prodAmpIndicesMatchingPattern(waveNamePattern);
```

Comment:
After a discussion with @legordian and @bgrube, I implemented the solution, suggested by @suhlatwork. 
The performance impact should hopefully not be to large, as the amplitude-indices-wave-index mapping is done in the same way in the spinDensityMatrixElem member functions. 
However, this **changes the interface of the intensityErr member function**, as the argument is no longer an amplitude-name regex but a wave-name regex, which makes it consistent with the intensity member function.